src/provable/provable-generic.ts

```diff
@@ -0,0 +1,306 @@
+import {
+  GenericHashInput,
```

Comment:
this file has the new generic version of the `provable` function. It's integrated into snarkyjs by this PR, but doesn't need to be reviewed in its entirety because it is practically the same code copied from `src/lib/circuit-value.ts`, just made generic.